install_config/upgrading/blue_green_deployments.adoc

```diff
@@ -0,0 +1,157 @@
+blue nodes are disabled so that no new pods are run on them.
+
+----
+# oadm manage-node --schedulable=true --selector=color=green
+# oadm manage-node --schedulable=false --selector=color=blue
+----
+
+A common practice is to scale the registry and router pods until they are
+migrated to the green nodes. For these pods, a _canary_ deployment approach is
+commonly used. Scaling them up will make them immediately active on the new
+nodes. Pointing the deployment configuration to the new image initiates a
+rolling update. However, because of node anti-affinity, and the fact that the
+blue nodes are still unschedulable, the deployments to the old nodes will fail.
+At this point, the registry and router deployments can be scaled down to the
+original number of pods. At any given point, the original number of pods is
+still available so no capacity is lost.
+
+[[manual-upgrades-warming-the-new-nodes]]
+== Warming the New Nodes
+
+In order for pods to be migrated from the blue environment to the green, the
+images must be pulled. Network latency and load on the registry can cause delays
+if there is not sufficient capacity built in to the environment. Often, the best
+way to minimize impact to the running system is to trigger new pod deployments that
+will land on the new nodes. Accomplish this by importing new *ImageStreams*.
+
+A major release of {product-title} is the motivation for a
+blue-green deployment. At that time, new *ImageStreams* become
+available for users of Source-to-Image. Upon import, any builds or deployments
+configured with *ImageChangeTriggers* are automatically created.
+
+It is important to realize that this process can trigger a large number of
+builds. The good news is that the builds are performed on the green nodes and,
+therefore, do not impact any traffic on the blue deployment.
+
+Monitor build progress:
+
+----
+$ oc get events -w --all-namespaces
+----
+
+In large environments, builds rarely completely stop. However, you should see a
+large increase and decrease caused by the administrative import.
+
+Another benefit of triggering the builds is that it does a fairly good job of
+fetching the majority of the ancillary images to all nodes such as the various
+build images, the pod infrastructure image, and deployers. Everything else can
+be moved over using node evacuation and will proceed more quickly as a result.
+
+[[manual-upgrades-node-evacuation]]
+== Node Evacuation
+
+For larger deployments, it is possible to have other labels that help
+determine how evacuation can be coordinated. The most conservative approach
+for avoiding downtime is to evacuate one node at a time. If services are
+composed of pods using zone anti-affinity, then an entire zone can be
+evacuated at once. It is important to ensure that the storage volumes used are
+available in the new zone as this detail can vary among cloud providers.
+
+In OpenShift Enterprise 3.2 and later, a node evacuation is triggered whenever
```

Comment:
Maybe split this using ifdefs and {product-title}s here so that it comes out saying "OpenShift Enterprise 3.2" for enterprise branches, and "OpenShift Origin 1.2" for origin.